ktor-server/ktor-server-core/jvm/src/io/ktor/response/ApplicationResponseFunctions.kt

```diff
@@ -12,22 +12,53 @@ import io.ktor.http.content.*
 import io.ktor.util.*
 import io.ktor.utils.io.*
 import java.io.*
+import kotlin.reflect.*
 
 /**
  * Sends a [message] as a response
  */
-@Suppress("NOTHING_TO_INLINE")
+@OptIn(ExperimentalStdlibApi::class)
+@JvmName("respondWithType")
+public suspend inline fun <reified T> ApplicationCall.respond(message: T) {
```

Comment:
Unlike the old respond function, this one provides a way to respond with `null`, including with `T = Nothing?`. Not sure if we need to allow it. If yes, we need to add a test for this

Reply:
Good point. Limited `T` to `T : Any`